### PR TITLE
Added disco3 parser and tests

### PR DESCRIPTION
**Pull request by aitordiaz**

### CODE REVIEW

File: src/averell/readers/disco3.py [hunk 1]
```diff
@@ -1,9 +1,135 @@
+import re
+import xml.etree.ElementTree as ETree
 from pathlib import Path
 
 XML_PATH = Path("tei") / "all-periods-per-author"
```
Comment (alvp) [inline src/averell/readers/disco3.py]: Missing newline before function declaration

File: src/averell/readers/disco3.py [hunk 2]
```diff
@@ -1,9 +1,135 @@
...
+                    'poem_type': line_group.get("type"),
+                    'author': author,
+                    'language': language,
+                })
+                id_list.append(work_dict["poem_id"])
+                alt_title = root.find(
+                    f".//*[@resource='disco:{sonnet_id}']"
+                    + "/{http://www.tei-c.org/ns/1.0}title[@property='dc:alternative']")
+                if alt_title is not None:
+                    alt_title = re.sub(r"[\n ]+", " ",
```
Comment (alvp) [inline src/averell/readers/disco3.py]: Better if you do 
`alt_title = re.sub(
                        r"[\n ]+", " ",`

File: src/averell/readers/disco3.py [hunk 3]
```diff
@@ -1,9 +1,135 @@
...
+                        'rhyme_pattern': stanza_rhyme_pattern,
+                        'stanza_text': "\n".join(stanza_text),
+                        'lines': line_list,
+                    })
+                    stanza_list.append(stanza_dict)
+
+            work_dict.update({'stanzas': stanza_list})
+            sonnet_list.append(work_dict)
+    # End line groups
+
```
Comment (alvp) [inline src/averell/readers/disco3.py]: Remove newline please

File: src/averell/readers/disco3.py [hunk 4]
```diff
@@ -1,9 +1,135 @@
...
+    if is_ensemble:
+        id_list.sort()
+        for sonnet in sonnet_list:
+            sonnet.update(
+                {"work_number": str(id_list.index(sonnet['poem_id']) + 1)})
+
```
Comment (alvp) [inline src/averell/readers/disco3.py]: Remove newline

File: src/averell/readers/disco3.py (hunk 1 above)
Comment (aitordiaz) [reply src/averell/readers/disco3.py]: Added newline

File: src/averell/readers/disco3.py (hunk 2 above)
Comment (aitordiaz) [reply src/averell/readers/disco3.py]: Done

File: src/averell/readers/disco3.py (hunk 3 above)
Comment (aitordiaz) [reply src/averell/readers/disco3.py]: Done

File: src/averell/readers/disco3.py (hunk 4 above)
Comment (aitordiaz) [reply src/averell/readers/disco3.py]: Done

**Pull request merged**